### PR TITLE
Adding BottomNavigation functionality

### PR DESCRIPTION
**Pull request by brownyte**
Could you guys offer some suggestions on how I could best move this code out of the MainActivity so that it is setup in the MVP architecture? In trying to figure this out myself, I came across this comment that said:

"For that scenario, you have a few options.
In general, from an MVP perspective, I treat the Fragment as the View and the Activity as just some auxiliary piece of the view. Presenter shouldn't care/know about what comprises the view.
So, you can solve this by enlisting a listener interface that the Activity implements and whenever your Presenter needs to update the toolbar:
Presenter calls view.updateToolbar() ->
Fragment receives event and forwards event to "ToolbarListener" (which is the Activity) eventListener.updateToolbar() ->
Activity receives the event and updates the Toolbar UI."

Does this mean that the MainActivity should have a Presenter associated with it? Should there be a FragmentManager util class that the various Fragments can call so that they can place their Fragment into the Frame container which will house the Fragments (which is in the xml file of activity_main)? This is really a headscratcher for me.

### CODE REVIEW

**Comment by chrsbell** [comment]
Want to hear what Joel and everyone thinks, yeah the way I handled navigation inside an Activity was to have the presenter for the activity but keeping all android/UI variables and usages in the Activity/Fragment like you mentioned. I would set the view for each presenter inside `onCreateView` for fragments and `onCreate` for activities. So I might have done something similar to this in MainActivity:

```
class MainActivity : AppCompatActivity() {
    private lateinit var bottomNavView : BottomNavigationView
    lateinit var binding: ActivityMainBinding
    private lateinit var landingPageFragment: LandingPageFragment
    private lateinit var profileFragment: ProfileFragment
    private lateinit var settingsFragment: SettingsFragment

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        bottomNavView = binding.bottomNavigationView

        landingPageFragment = LandingPageFragment()
        profileFragment = ProfileFragment()
        settingsFragment = SettingsFragment()

        // setup presenter
        presenter = MainPresenter()
        presenter.setView(this)
        
        presenter.navigate(R.id.landingPageFragment) //setting initial view to landing page as you only get here after bypassing login/register screens

        bottomNavView.setOnItemSelectedListener {
            presenter.navigate(it.itemId)
        }
    }

    private fun handleNavigateById(desiredFragmentId: Int) {
        val desiredFragment = when (desiredFragmentId) {
            R.id.landingPageFragment -> landingPageFragment
            R.id.profileFragment -> profileFragment
            R.id.settingsFragment -> settingsFragment
            else -> null
        }
        if (desiredFragment != null) {
            supportFragmentManager.beginTransaction().apply {
                replace(R.id.frame, desiredFragment)
                commit()
            }
        }
    }
}

// inside presenter
// private fun navigate(id: Int) = view.handleNavigateById(id)
```

Not sure about the FragmentManager part sorry

**Comment by brownyte** [comment]
This is really fascinating. I will discuss this in my one on one with Joel later. Thank you so much for your insight, Chris. I really appreciate having the opportunity to learn from you.

**Comment by chrsbell** [comment]
> This is really fascinating. I will discuss this in my one on one with Joel later. Thank you so much for your insight, Chris. I really appreciate having the opportunity to learn from you.

Sounds good! I could be very wrong here haha.

**Review by chrsbell — APPROVED** [review]
Looks good!

File: app/src/main/java/com/example/voyagerx/MainActivity.kt [hunk 1]
```diff
@@ -2,12 +2,41 @@ package com.example.voyagerx
...
+
+        bottomNavView.setOnItemSelectedListener {
+            when (it.itemId) {
+                R.id.landingPageFragment -> changeFragmentView(landingPageFragment)
+                R.id.profileFragment -> changeFragmentView(profileFragment)
+                R.id.settingsFragment -> changeFragmentView(settingsFragment)
+            }
+            true
+        }
     }
```
Comment (Celinarabe) [inline app/src/main/java/com/example/voyagerx/MainActivity.kt:35]: I think you can remove the assignments and create the fragments directly. That way we're only creating fragments when we need to.

```suggestion

        changeFragmentView(LandingPageFragment()) //setting initial view to landing page as you only get here after bypassing login/register screens

        bottomNavView.setOnItemSelectedListener {
            when (it.itemId) {
                R.id.landingPageFragment -> changeFragmentView(LandingPageFragment())
                R.id.profileFragment -> changeFragmentView(ProfileFragment())
                R.id.settingsFragment -> changeFragmentView(SettingsFragment())
            }
            true
        }
    }
```

**Review by Celinarabe — COMMENTED** [review]
> Does this mean that the MainActivity should have a Presenter associated with it? Should there be a FragmentManager util class that the various Fragments can call so that they can place their Fragment into the Frame container which will house the Fragments (which is in the xml file of activity_main)?

Also regarding your question, I think the FragmentManager is the way to go! From my minimal research, it seems like supportFragmentManager was built for the exact use case of swapping out fragments. I think this is how you decided to implement it in the end?

File: app/src/main/java/com/example/voyagerx/MainActivity.kt (hunk 1 above)
Comment (joelc1225) [reply app/src/main/java/com/example/voyagerx/MainActivity.kt:35]: Great suggestion

**Review by joelc1225 — CHANGES_REQUESTED** [review]
After implementing Celina's suggestion, this LGTM (Looks good to me). 

Also, @chrsbell's code snippet looks clean and makes sense to me 👍

File: app/src/main/java/com/example/voyagerx/MainActivity.kt (hunk 1 above)
Comment (brownyte) [reply app/src/main/java/com/example/voyagerx/MainActivity.kt]: Thanks, Celina! I implemented this change.

**Comment by brownyte** [comment]
> me container which will house the Fragments (which is in the xml file of activity_main)?

You are correct. I was thinking about your Details Fragment and my Edit Profile Fragment reusing the same FragmentManager that the BottomNavigation fragments are using to replace their view in. I thought if they used the same FragmentManager that it should be a separate class (to get that code out of MainActivity), but I am not sure that makes any sense (especially because we have decided to leave this code in the MainActivity). Thanks for your candor and insight!

**Review by joelc1225 — APPROVED** [review]

**Pull request merged**